### PR TITLE
Create !parking

### PR DESCRIPTION
**Pull request by bradleysigma**
Creates `!parking`, which lists casual parking available, and `!parking all`, which lists casual and permit parking available.
image
Implements `parking.js` from `unimplemented_commands`.

### CODE REVIEW

**Comment by jdcaperon** [comment]
If you put the parking text in triple quotes does it condense the information a bit better? Just a formatting recommendation.

**Comment by bradleysigma** [comment]
> If you put the parking text in triple quotes does it condense the information a bit better? Just a formatting recommendation.

Which text (what line[s])?

File: uqcsbot/scripts/parking.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+             "P3": "P3 - Multi-Level West (Staff)", "P4": "P4 - Multi-Level East (Staff)",
+             "P6": "P6 - Hartley Teakle (14P Hourly)", "P7": "P7 - DustBowl (14P Daily)",
+             "P7 UC": "P7 - Keith Street (14P Daily Capped)",
+             "P8 L1": "P8 - Athletics Basement (14P Daily)",
+             "P8 L2": "P8 - Athletics Roof (14P Daily)", "P9": "P9 - Boatshed (14P Daily)",
+             "P10": "P10 - UQ Centre & Playing Fields (14P Daily/14P Daily Capped)",
+             "P11 L1": "P11 - Conifer Knoll Lower (Staff)",
+             "P11 L2": "P11 - Conifer Knoll Upper (Staff)",
+             "P11 L3": "P11 - Conifer Knoll Roof (14P Daily Restricted)"}
+    catagory = lambda x: "No" if x == "FULL" else "Few" if x == "NEARLY FULL" else x
```
Comment (nicklambourne) [inline uqcsbot/scripts/parking.py]: sp. category

File: uqcsbot/scripts/parking.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+    # find parks
+    areas = findall(r"<tr>\W*<td class='zone'>(.*)<\/td>\W*<td class='.*\n?'>(.*)\n?<\/td>" +
+                    r"\W*<td class='.*\n?'>(.*)\n?<\/td>\W*<\/tr>", str(data))
+    for a in areas:
```
Comment (nicklambourne) [inline uqcsbot/scripts/parking.py]: `for area in areas` please

File: uqcsbot/scripts/parking.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+from re import findall
+from typing import Union
+
+
+def get_pf_parking_data() -> Union[str, None]:
+    """
+    Returns a parking HTML document from the UQ P&F website
+    """
+    try:
+        data = urlopen("https://pg.pf.uq.edu.au/").read().decode("utf-8")
```
Comment (nicklambourne) [inline uqcsbot/scripts/parking.py]: requests library may have been easier, just a note for next time

File: uqcsbot/scripts/parking.py (hunk 1 above)
Comment (bradleysigma) [reply uqcsbot/scripts/parking.py]: Fixed.

File: uqcsbot/scripts/parking.py (hunk 2 above)
Comment (bradleysigma) [reply uqcsbot/scripts/parking.py]: Fixed.

File: uqcsbot/scripts/parking.py (hunk 3 above)
Comment (bradleysigma) [reply uqcsbot/scripts/parking.py]: Fixed.

Comment (bradleysigma) [reply uqcsbot/scripts/parking.py]: (Though this still needed for `test_parking.py`.)

**Pull request merged**